### DISCUSSION TITLE
regex in authorization file in topic field

### DISCUSSION THREAD

**Discussion by rickou** [emqx/emqx#14392] [Ideas]
it could be useful to have the poissibility to do a regex in the topic field in the authorization file.
for my request, i would like my "master" device to restrict publishing into topics like:
gb/GBOX[0-9]{6}/config
today it seems i'm limited to something like : gb/+/config 
but it is less restrictive.

**Comment by id** [comment]
You can certainly use regex expression in acl file: https://docs.emqx.com/en/emqx/latest/access-control/authz/file.html

**Reply by rickou** [reply]
if i'm right i can't use it in the topic field...

**Reply by id** [reply]
Right, sorry, my bad.

**Comment by zmstone** [comment]
EMQX Enterprise's schema validation feature can check topic and payload.

https://docs.emqx.com/en/emqx/latest/data-integration/schema-validation.html#schema-validation

Example:
```
schema_validation {
  validations = [
    {
      checks = [
        {
          sql = "select * where regex_match(topic, '^gb/GBOX[0-9]{6}/config$')"
          type = sql
        }
      ]
      description = ""
      failure_action = drop
      log_failure {level = warning}
      name = "regex-check-topic"
      strategy = all_pass
      topics = [
        "gb/#"
      ]
    }
  ]
}
```
I am however, not sure how practically useful it is to check just the topic pattern, maybe what you want is to check the client can also be authenticated/authorized using the pattern `GBOX[0-9]{6}`,
client attribute is likely a better fit for such checks. see the other discussion: https://github.com/emqx/emqx/discussions/14391#discussioncomment-11541614
